src/renderer/lib/errors.js

```diff
@@ -1,15 +1,54 @@
-module.exports = {
-  UnplayableTorrentError,
-  UnplayableFileError
+const ExtendableError = require('es6-error')
+
+/* Generic errors */
+
+class CastingError extends ExtendableError {}
+class PlaybackError extends ExtendableError {}
+class SoundError extends ExtendableError {}
+class TorrentError extends ExtendableError {}
+
+/* Playback */
+
+class UnplayableTorrentError extends PlaybackError {
+  constructor () { super('Can\'t play any files in torrent') }
+}
+
+class UnplayableFileError extends PlaybackError {
+  constructor () { super('Can\'t play that file') }
+}
+
+class PlaybackTimedOutError extends PlaybackError {
+  constructor () { super('Playback timed out. Try again.') }
 }
 
-function UnplayableTorrentError () {
-  this.message = 'Can\'t play any files in torrent'
+/* Sound */
+
+class InvalidSoundNameError extends SoundError {
+  constructor (name) { super(`Invalid sound name: ${name}`) }
 }
 
-function UnplayableFileError () {
-  this.message = 'Can\'t play that file'
+/* Torrent */
+
+class TorrentKeyNotFoundError extends TorrentError {
+  constructor (torrentKey) { super(`Can't resolve torrent key ${torrentKey}`) }
 }
 
-UnplayableTorrentError.prototype = Error
-UnplayableFileError.prototype = Error
+class InvalidTorrentError extends TorrentError {}
+
+/* Miscellaneous */
+
+class IllegalArgumentError extends ExtendableError {}
+
+module.exports = {
```

Comment:
Note on consistency. The `exports` statement can't be on the top like in the other files due to class declarations being non-hosted in ES6.